src/components/PeopleTable.tsx

```diff
@@ -1,644 +1,110 @@
+import PersonLink from './PersonLink';
+import { Person } from './../types/Person';
+import { PeopleMap } from '../types/PeopleMap';
+import { SearchLink } from './SearchLink';
+import { SearchParams } from '../utils/searchHelper';
+import React from 'react';
+
+type TableColumn = {
+  column: string;
+  sortable: boolean;
+};
+
+type TableColumns = {
+  [property in keyof Partial<Person>]: TableColumn;
+};
+
+const TABLE_COLUMNS: TableColumns = {
+  name: { column: 'Name', sortable: true },
+  sex: { column: 'Sex', sortable: true },
+  born: { column: 'Born', sortable: true },
+  died: { column: 'Died', sortable: true },
+  motherName: { column: 'Mother', sortable: false },
+  fatherName: { column: 'Father', sortable: false },
+};
+
 /* eslint-disable jsx-a11y/control-has-associated-label */
-export const PeopleTable = () => {
+export const PeopleTable = ({
+  people,
+  selectedPerson,
+  peopleMap,
+  onArrowClick,
+  sortColumn,
+  sortDirection,
+}: {
+  people: Person[];
+  selectedPerson?: string | null;
+  peopleMap: PeopleMap;
+  onArrowClick: (column: string) => SearchParams;
+  sortColumn: keyof Person | null;
+  sortDirection: boolean;
+}) => {
+  const shouldHighlightUser = (slug: string) =>
+    selectedPerson === slug ? 'has-background-warning' : '';
+  // TODO class needs to receive the sort
+
   return (
     <table
       data-cy="peopleTable"
       className="table is-striped is-hoverable is-narrow is-fullwidth"
     >
       <thead>
         <tr>
-          <th>
-            <span className="is-flex is-flex-wrap-nowrap">
-              Name
-              <a href="#/people?sort=name">
-                <span className="icon">
-                  <i className="fas fa-sort" />
-                </span>
-              </a>
-            </span>
-          </th>
-
-          <th>
-            <span className="is-flex is-flex-wrap-nowrap">
-              Sex
-              <a href="#/people?sort=sex">
-                <span className="icon">
-                  <i className="fas fa-sort" />
-                </span>
-              </a>
-            </span>
-          </th>
-
-          <th>
-            <span className="is-flex is-flex-wrap-nowrap">
-              Born
-              <a href="#/people?sort=born&amp;order=desc">
-                <span className="icon">
-                  <i className="fas fa-sort-up" />
-                </span>
-              </a>
-            </span>
-          </th>
-
-          <th>
-            <span className="is-flex is-flex-wrap-nowrap">
-              Died
-              <a href="#/people?sort=died">
-                <span className="icon">
-                  <i className="fas fa-sort" />
+          {Object.entries(TABLE_COLUMNS).map(([key, { column, sortable }]) => {
+            const isSortedColumn = sortColumn === key;
+
+            const sortIcon = isSortedColumn
+              ? sortDirection
+                ? 'fa-sort-down'
+                : 'fa-sort-up'
+              : 'fa-sort';
```

Comment:
The logic for determining the sort icon is correct. However, ensure that the `onArrowClick` function correctly updates the sorting state to reflect the current sort direction.